addons/web_editor/models/ir_qweb.py

```diff
@@ -53,9 +53,11 @@ def _compile_directive_snippet(self, el, options):
         view_id = View.get_view_id(el.attrib.get('t-call'))
         name = View.browse(view_id).name
         thumbnail = el.attrib.pop('t-thumbnail', "oe-thumbnail")
-        div = u'<div name="%s" data-oe-type="snippet" data-oe-thumbnail="%s">' % (
+        forbid_sanitize = el.attrib.get('t-forbid-sanitize')
```

Comment:
One issue with this is it requires updating the views in stable. Maybe the information could be inferred instead? e.g. run `html_sanitize` on the snippet and check the result?

It would be more expensive tho. But assuming that's for the snippets sidebar (is it?) that might be an acceptable trade. It might be a good idea to cache the "query results" in that case (e.g. have a global map of snippet => sanitization flags)

Reply:
@xmo-odoo Answers here: https://github.com/odoo/odoo/pull/89402#issuecomment-1106362854

1) We cannot detect via html_sanitize, the main case we are trying to solve is the embed code snippet... which is just an empty div that the user can fill **during edition**. That "div" has to be *marked* unsanitizable. We could do both though, but not sure it is worth the trouble.

2) As indicated, we'll mark the unsanitizable snippets via JS in stable too. Agreed, that PR should be in draft :wink:

Reply:
Made all snippets containing `<form>` automatically marked as forbidden in sanitized fields.